pallets/democracy-mbm/src/lib.rs

```diff
@@ -0,0 +1,311 @@
+    fn post_upgrade(state: Vec<u8>) -> Result<(), sp_runtime::TryRuntimeError> {
+        try_runtime::post_upgrade_body::<T>(state)
+    }
+}
+
+impl<T: pallet_democracy::Config + frame_system::Config, W: weights::WeightInfo>
+    DemocracyMigrationV1ToV2<T, W>
+{
+    fn required_weight(step: &MigrationState<T>) -> Weight {
+        match step {
+            MigrationState::ReferendumInfo(_) => W::migration_referendum_info(),
+            MigrationState::StartingVotingOf | MigrationState::VotingOf(_) => {
+                W::migration_voting_of()
+            }
+            MigrationState::Finished => Weight::zero(),
+        }
+    }
+
+    fn migrate_referendum_info(
+        maybe_last_key: Option<&ReferendumIndex>,
+        current_block_number: u32,
+    ) -> StepResultOf<T> {
+        let mut iter = if let Some(last_key) = maybe_last_key {
+            ReferendumInfoOf::<T>::iter_from(ReferendumInfoOf::<T>::hashed_key_for(last_key))
+        } else {
+            ReferendumInfoOf::<T>::iter()
+        };
+
+        if let Some((last_key, mut ref_info)) = iter.next() {
+            match ref_info {
+                ReferendumInfo::Ongoing(ref mut status) => {
+                    // Double the blocks of the delay period
+                    status.delay = status
+                        .delay
+                        .saturated_into::<u32>()
+                        .saturating_mul(2)
+                        .into();
+
+                    // For the end time:
+                    // 1. Calculate remaining blocks until the original end
+                    let remaining_blocks = status
+                        .end
+                        .saturated_into::<u32>()
+                        .saturating_sub(current_block_number);
+
+                    // 2. Double the remaining blocks
+                    let doubled_remaining = remaining_blocks.saturating_mul(2);
+
+                    // 3. Add it to the current block number to get the new end
+                    status.end = current_block_number
+                        .saturating_add(doubled_remaining)
+                        .into();
+                }
+                ReferendumInfo::Finished { .. } => {
+                    // continue;
+                }
+            }
+
+            ReferendumInfoOf::<T>::insert(&last_key, ref_info.clone());
+
+            MigrationState::ReferendumInfo(last_key)
+        } else {
+            MigrationState::StartingVotingOf
+        }
+    }
+
+    fn migrate_voting_of(
+        maybe_last_key: Option<&T::AccountId>,
+        current_block_number: u32,
+    ) -> StepResultOf<T> {
+        let mut iter = if let Some(last_key) = maybe_last_key {
+            VotingOf::<T>::iter_from(VotingOf::<T>::hashed_key_for(last_key))
+        } else {
+            VotingOf::<T>::iter()
+        };
+
+        if let Some((last_key, mut voting)) = iter.next() {
+            match &mut voting {
+                Voting::Direct { prior, .. } => {
```

Comment:
As dino said, match branches can be combined like this
```suggestion
                Voting::Direct { prior, .. } | Voting::Delegating { prior, .. } => {
```